desk/app/chat.hoon

```diff
@@ -902,6 +905,43 @@
       !>(`a-channels:d`[%channel [%chat flag] %post u.command])
     `[%pass /migrate %agent [our.bowl %channels] %poke cage]
   ::
+  ++  trim
+    =-  =.  old-chats  -  cor
+    ^-  (map flag:two:old:c chat:two:old:c)
+    %-  ~(run by old-chats)
+    |=  old-chat=chat:two:old:c
+    =/  citations=(set [ship time])
+      %-  sy
+      ^-  (list [ship time])
+      %-  zing
+      ^-  (list (list [ship time]))
+      %+  murn  (tap:on:writs:old wit.pact.old-chat)
+      |=  [=time =writ:old]
+      ^-  (unit (list [ship ^time]))
+      ::  return citer message and cited message
+      ?.  =(our.bowl author.writ)  ~
+      =/  cite-targets=(list [ship ^time])
+        ?.  ?=(%story -.content.writ)  ~
+        %+  murn  p.p.content.writ
+        |=  =block:old
+        ^-  (unit [ship ^time])
+        ?.  ?=([%cite %chan [%chat *] *] block)  ~
+        ?.  ?=([%msg @ @ ~] wer.cite.block)  ~
+        =/  who  (slaw %p i.t.wer.cite.block)
+        ?~  who  ~
+        =/  tim  (slaw %ud i.t.t.wer.cite.block)
+        ?~  tim  ~
+        `[u.who u.tim]
+      ?~  cite-targets
+        ~
+      `[[author.writ time] cite-targets]
+    %=  old-chat  wit.pact
+      %-  malt
+      %+  murn  (tap:on:writs:old wit.pact.old-chat)
+      |=  [=time =writ:old]
+      ?.  (~(has in citations) [author.writ time])  ~
+      `[time writ]
```

Comment:
I think you want to check the `dex` since the `[ship time]` is an `id` the time won't match exactly what's in `wit` since those times are different.